Jobsway2goMvc/Controllers/PostController.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using Jobsway2goMvc.Data;
+using Jobsway2goMvc.Models;
+using System.Security.Claims;
+
+namespace Jobsway2goMvc.Controllers
+{
+    public class PostController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public PostController(ApplicationDbContext context, IHttpContextAccessor httpContextAccessor)
+        {
+            _context = context;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        // GET: Post
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.Posts.ToListAsync());
+        }
+
+        // GET: Post/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null || _context.Posts == null)
+            {
+                return NotFound();
+            }
+
+            var post = await _context.Posts
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (post == null)
+            {
+                return NotFound();
+            }
+
+            return View(post);
+        }
+
+        private ApplicationUser GetApplicationUser(ClaimsPrincipal principal)
+        {
+            var userId = principal.FindFirstValue(ClaimTypes.NameIdentifier);
+
+            var user = _context.Users.FirstOrDefault(u => u.Id == userId);
+            
+            return user;
+        }
+
+        // GET: Post/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: Post/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,Title,Description,CreatedAtUTC,Type")] Post post)
+        {
+            ModelState.Remove("CreatedBy");
+            if (ModelState.IsValid)
+            {
+                var user = _httpContextAccessor.HttpContext.User;
+                var applicationUser = GetApplicationUser(user);
+                post.CreatedBy = applicationUser;
+                _context.Add(post);
```

Comment:
Remove `var applicationUser = GetApplicationUser(user);` and do it directly `post.CreatedBy = GetApplicationUser(user);`